File: src/main/java/com/yaojie/modules/medicine/entity/Medicine.java
package com.yaojie.modules.medicine.entity;

import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class Medicine {

    private Long id;
    private String medicineCode;
    private String medicineName;
    private Long categoryId;
    private String specification;
    private String unit;
    private String manufacturer;
    private Long supplierId;
    private BigDecimal purchasePrice;
    private BigDecimal salePrice;
    private Integer safeStock;
    private Integer expiryWarningDays;
    private Integer isControlled;
    private Integer status;
    private String remark;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/types/medicine.ts
export interface MedicineItem {
  id: number
  medicineCode: string
  medicineName: string
  categoryId: number
  categoryName?: string
  specification?: string
  unit: string
  manufacturer?: string
  supplierId?: number
  supplierName?: string
  purchasePrice: number
  salePrice: number
  safeStock: number
  expiryWarningDays: number
  isControlled: number
  status: number
  remark?: string
  createdAt?: string
  updatedAt?: string
}

export interface MedicineQuery {
  keyword?: string
  categoryId?: number
  supplierId?: number
  isControlled?: number
  status?: number
}

export interface MedicineCategory {
  id: number
  categoryName: string
  categoryCode: string
  remark?: string
}

export interface SupplierItem {
  id: number
  supplierName: string
  contactPerson?: string
  phone?: string
  address?: string
  status: number
}

export interface MedicineForm {
  medicineCode: string
  medicineName: string
  categoryId: number | undefined
  specification: string
  unit: string
  manufacturer: string
  supplierId: number | undefined
  purchasePrice: number
  salePrice: number
  safeStock: number
  expiryWarningDays: number
  isControlled: number
  status: number
  remark: string
}
